Sources/JSON.swift

```diff
@@ -12,13 +12,57 @@ public enum JSON {
     case null
 }
 
+extension JSON.Number {
+    // TODO: We could make these public, but people might prefer not to.
+    private var double: Double {
+        switch self {
+        case let .double(d):
+            return d
+        case let .integer(i):
+            return Double(i)
+        case let .unsignedInteger(u):
+            return Double(u)
+        }
+    }
+
+    private var int: Int? {
+        switch self {
+        case let .double(d):
+            return Int(d)
+        case let .integer(i):
+            return i
+        case let .unsignedInteger(u):
+            if u < UInt(Int.max) {
+                return Int(u)
+            } else {
+                return nil
+            }
+        }
+    }
+
+    private var uint: UInt? {
+        switch self {
+        case let .double(d) where d >= 0:
+            return UInt(d)
```

Comment:
So if you'd be comparing `UInt(2)` and `Double(2.1)`, it'd return `true`. That doesn't seem right